zetaclient/chains/evm/observer/inbound.go

```diff
@@ -41,37 +43,38 @@ func (ob *Observer) WatchInbound(ctx context.Context) error {
 		return err
 	}
 
-	ticker, err := clienttypes.NewDynamicTicker(
-		fmt.Sprintf("EVM_WatchInbound_%d", ob.Chain().ChainId),
-		ob.GetChainParams().InboundTicker,
-	)
-	if err != nil {
-		ob.Logger().Inbound.Error().Err(err).Msg("error creating ticker")
-		return err
-	}
-	defer ticker.Stop()
-
 	ob.Logger().Inbound.Info().Msgf("WatchInbound started for chain %d", ob.Chain().ChainId)
-	sampledLogger := ob.Logger().Inbound.Sample(&zerolog.BasicSampler{N: 10})
 
-	for {
-		select {
-		case <-ticker.C():
-			if !app.IsInboundObservationEnabled(ob.GetChainParams()) {
-				sampledLogger.Info().
-					Msgf("WatchInbound: inbound observation is disabled for chain %d", ob.Chain().ChainId)
-				continue
-			}
-			err := ob.ObserveInbound(ctx, sampledLogger)
-			if err != nil {
-				ob.Logger().Inbound.Err(err).Msg("WatchInbound: observeInbound error")
-			}
-			ticker.UpdateInterval(ob.GetChainParams().InboundTicker, ob.Logger().Inbound)
-		case <-ob.StopChannel():
-			ob.Logger().Inbound.Info().Msgf("WatchInbound stopped for chain %d", ob.Chain().ChainId)
+	var (
+		logger   = ob.Logger().Inbound.Sample(&zerolog.BasicSampler{N: 10})
+		interval = ticker.SecondsFromUint64(ob.GetChainParams().InboundTicker)
+	)
+
+	t := ticker.New(interval, func(ctx context.Context, t *ticker.Ticker) error {
```

Comment:
What about define the function separately instead of use a literal?

Reply:
https://github.com/zeta-chain/node/pull/2617/commits/5cb4839f43770a0753a04b09ae241bfb3bf67e33